frame/bounties/src/lib.rs

```diff
@@ -653,12 +780,596 @@ decl_module! {
 					},
 					_ => return Err(Error::<T>::UnexpectedStatus.into()),
 				}
-
+				// TODO :: Have to recheck
+				// Extend expiry for active subbounty
+				// if bounty.active_subbounty.len() > 0 {
+				// 	bounty.active_subbounty
+				// 	.iter()
+				// 	.for_each(|subbounty_id| {
+				// 		let _ = Self::impl_extend_subbounty_expiry(&signer, bounty_id, *subbounty_id, remark.clone());
+				// 	});
+				// }
 				Ok(())
 			})?;
 
 			Self::deposit_event(Event::<T>::BountyExtended(bounty_id));
 		}
+
+		/// Add a new subbounty.
+		///
+		/// The dispatch origin for this call must be curator.
+		/// Bounty must me in "active" state.
+		///
+		/// Subbouty gets added successfully & fund gets reserved, if bounty has enough fund.
+		/// else call get failed.
+		///
+		/// Upperbount to maximum number of subbounties that can be added is
+		/// managed via runtime trait config 'MaxSubBountyCount'.
+		///
+		/// Payment: `TipReportDepositBase` will be reserved from the origin account, as well as
+		/// `DataDepositPerByte` for each byte in `reason`. It will be unreserved upon approval,
+		/// or slashed when rejected.
+		///
+		/// if call is success, state of subbounty is moved to "Approved" state.
+		/// And later moved to "Funded" state as part of "spend_fund()" callback.
+		///
+		/// - `bounty_id`: Bounty ID for which subbounty to be added.
+		/// - `value`: Value for executing the proposal.
+		/// - `description`: Text description for the subbounty.
+		#[weight = 10_000]
+		fn add_subbounty(origin,
+			#[compact] bounty_id: BountyIndex,
+			value: BalanceOf<T>,
+			description: Vec<u8>,
+		)
+		{
+			let signer = ensure_signed(origin)?;
+
+			Bounties::<T>::try_mutate_exists(bounty_id, |maybe_bounty| -> DispatchResult {
+				let mut bounty = maybe_bounty.as_mut().ok_or(Error::<T>::InvalidIndex)?;
+				match bounty.status {
+					BountyStatus::Active {
+						ref curator,
+						..
+					} => {
+						ensure!(signer == *curator, Error::<T>::RequireCurator);
+						// Verify the arguments
+						ensure!(description.len() <=
+							T::MaximumReasonLength::get() as usize,
+							Error::<T>::ReasonTooBig,
+						);
+						ensure!(value >= T::BountyValueMinimum::get(),
+							Error::<T>::InvalidValue,
+						);
+						ensure!((bounty.active_subbounty.len() as u32) <
+							T::MaxSubBountyCount::get() as u32,
+							Error::<T>::SubBountyMaxOverflow,
```

Comment:
Overflow is when you do addition and sum too large to fit, this is just TooManySubBounties

Reply:
Got it, replaced error code to "TooManySubBounties"